### ISSUE TITLE
Tips to know BEFORE ordering keyboard PCB!

### ISSUE THREAD

**Issue by gwideman** [marcoreps/5700A#1]
First -- big thanks to Marco Reps for creating this PCB design and putting it on GitHub!

We ordered some (from AllPCB.com), for use in a Fluke 5500A, which uses the same keyboard PCB as the 5700A. (You can check the parts list in the service manual.)

And we can confirm that they work nicely.

However, there are some points we learned that may either allow you to bypass replacing the PCB, or may influence you to modify it before ordering. And will give you a head-start on ordering other components you may need.

If you’re reading this, it’s likely you’re motivated by one or more keys that work only intermittently, or not at all.  And you’re hoping that the culprit is the crud on the PCB patterns that Marco shows in his video. 

[(https://www.youtube.com/watch?v=qhr6h7wuqo0)]

So our main recommendation is to investigate whether instead the fault may be with the rubbery conductive pads on the back of the rubber membrane key caps, which press down on the PCB patterns to make the connection. In our case, the conductive pad on our non-functioning key (the popular “OPR” [operate] button) had lost all conductivity. So replacing the keyboard PCB didn’t fix that problem, though it probably did improve the keyboard, and we don’t regret fitting it.

More on the non-conducting key pad later.

**Preparation**

Before starting, test the operation of all the keys to note which ones don’t work. 

Get the services manual. Google for the PDF applicable to your machine and download it. We noticed that for some models the PDF is available with or without schematics. 

If your machine is not a 5700 or 5500A, verify that it uses the same keyboard part number, which is:

Ref Designator A1
Description: Keyboard PCA
Fluke Stock No: 761049

**PCB order**

For what it’s worth, we chose standard 1.6mm board thickness (though the original is probably more like 2.5 to 3mm). This did not cause a problem with positioning relative to the key membrane. But it did change the position of the photointerruptor assembly slightly, but not enough to rub. You could always solder that assembly a little raised.

We chose Surface Finish = Immersion Gold (sometimes called ENIG), and selected 2.00 thickness of the gold finish rather than 1.0. 

**Dissassembly tips**

Check Marco’s video and the service manual for the main details. I’ve attached the exploded diagram of the front assembly below.

These are just some extra tips:

**Allen screws:** In removing the front handles and front panel assembly, you’ll need to remove a bunch of black cheesehead screws that take a tiny allen key. If possible get a long allen key, preferably one that fits a T-handle. It’s not fun trying to deal with those screws with a clumsy “swiss army knife” allen key collection.

**Bottom cover:** You will easily see you need to remove the top cover of the instrument. On the 5500A, it’s less obvious that you need to remove the bottom cover too. This is to gain access to the connector for the cable that feeds the voltage and current outputs (and their sense wires) to the front panel jacks. That connector will get yanked loose when you remove the front assembly, but is too short to reattach until the front assembly is screwed in place, at which point you can only access the connector through the bottom. 

That connector arrangement was changed on the 5700A (Marco’s video, 9:22), so bottom panel removal is not necessary.

The rest of the disassembly proceeds in straightforward fashion by unscrewing obviously-visible fasteners or releasing plastic clips.

To remove the keyboard PCB, you do need to unscrew the optical interruptor disc (Video 12:33). You’ll need to hold the knob from the front in order to apply sufficient torque to the screw that holds the disc.

**Testing the keys**

Marco’s video shows an example membrane keyboard at 11:55. This is not the actual membrane assembly in the unit, but uses the same principle of pressing a black conductive rubbery pad onto the PCB pattern.

So the task is to check the resistance of the rubbery pads on the malfunctioning keys, and compare to the resistance on the good keys. We found it best to push the key so that the rubber pad sticks out, then lay a couple of DMM test probes parallel to each other and flat across the pad, at a consistent distance apart. We got resistances in the range from a few k ohms to low 10k’s of ohms, depending on spacing.

We discovered that the non-functioning OPR key had infinite resistance, or over 1 Mohm anyway. We also noticed it had a shiny surface, where other pads were matte in appearance. We carefully used a Q-tip and some IPA to clean the surface of that pad, which then appeared matte like the others. But still no conductivity.

**Dealing with a dead key**

If you discover this before ordering the PCB, you may want to change plans.

Our unit did not have the optional Scope module installed, and will never get it. So we decided to put the SCOPE button into service as the new OPR button.

One way to do this would be to perform a little surgery on the PCB, to detach the SCOPE button PC pattern from the nets it’s connected to, and jumper it over to the existing OPR button’s PC pattern, running the wires on the back of the board.

However, there’s no easy way to detach the pattern for the SCOPE key from its existing nets without interrupting those nets going to other keys. And it’s difficult to remediate _that_ problem as you can’t run wires on the keypad side of the board because that would lift up the rubber membrane. 

_Idea for improvement of the PCB:_ Keep the through-holes at the sides of each key pattern, but for each one create a separate nearby through-hole to transmit the nets through the board. Obviously put a short link trace between the two through-holes; this can be cut if needed to isolate that key, so it can be rewired for another purpose.

Our workaround was to use a knife to cut out the OPR and SCOPE key caps from the rubber membrane, and exchange their positions. So now we have a working key at the OPR location, which is labeled SCOPE. The key caps are held in place well by the front panel with its snug-fitting openings.

This worked fine with the new PCB. We did not go back and test it with the old PCB. But it may well work, and if it does for you, obviously it will save replacing the keyboard PCB.

**Other component notes**

**IDC connector:** Marco’s video (12:40) shows an IDC plug and socket connecting the keyboard to the other board in the front panel assembly. However, in the 5500A we found that the ribbon cable was permanently connected to the board by a low-profile solder-on IDC fitting.  So the cable was not detachable at that end.

We did not want to unsolder that large of a connection, so we decided to install a new ribbon cable, and add a male header on the nw PCB.  However, we then saw there was not enough clearance for the height of that connector. So instead we installed a right-angle header, and that worked OK. (You may want to be extra careful to trim the cutoff end of the ribbon cable where it exits the IDC clamp, and points toward the PCB, to avoid any possibility of the exposed ends shorting to the board if they manage to wear  through the solder mask. Or add some robust tape between the IDC connector and the board.)

![Fluke_5500A_Keyboard_RA_IDC_connector](https://user-images.githubusercontent.com/1243021/97777876-44b49e80-1b30-11eb-8cd4-789d20c1c3e5.jpg)

**Side note on replacement cable:** this is a 2x20 pin 0.1 x 0.1“ pitch pattern, that works with IDC connectors that use cable with conductors on 0.05” centers. As such, it’s just like original PC IDE PATA cables, and such a cable works (we used one). However the newer mechanically compatible 2x20 PATA IDE “Ultra DMA“   cables that have 0.025”-pitch wires do not work. Those cables have interleaved ground wires between all the existing wires, to reduce crosstalk. Those intervening wires connect inside the innocent-looking but PATA-specific IDC connector to all the PATA ground pins, which means that all those PATA ground pins are shorted together. Obviously such a cable will not work for this keyboard job.

The opto-interruptor encoder assembly: This has two sets of 4 pins and must be transferred from the old board to the new. We were not able to unsolder these with solder wick or solder vaccuum, and had to use a hot air station to heat all the pins simultaneously.

**LEDs:** The LEDs are very low-profile, suspended in mid air in holes in the PCB, with legs sticking out sideways instead of inline with the LED direction. See video 10:43. (For some reason, this lead arrangement is described as “axial”, though it really seems radial to me.) We intended to just use ordinary 3mm through-hole LEDs with legs suitably bent, but there was insufficient clearance for the height of the LED plus any lead length before bending. (This is in the narrow part of the wedge shape of the keyboard assembly.) So we had to transfer the old LEDs over. That was a dicey job, as they are quite fragile (moreso on the 5500A I think.). So you may want to get some appropriate LEDs to install on the new board. This one looks good:  https://www.digikey.com/en/products/detail/lumex-opto-components-inc/SSL-LXA228SGC/270934 "LED GREEN CLEAR AXIAL T/H”. 

**PCB attachment screws:** In several places, traces on the initial version of Marco’s PCB run under the wide screw heads and are vulnerable to shorting adjacent traces. If that's the case on your board, insert some insulting washers, or some other insulating shim. [Edit: Marco wrote in a comment on this issue that he's updated the PCB design already to route those traces further away from the mounting holes. Noice!]

OK, I hope these notes help someone!

![5500A_FrontPanelAssembly](https://user-images.githubusercontent.com/1243021/97777900-6e6dc580-1b30-11eb-9483-ae4d205a393b.png)

**Comment by marcoreps** [comment]
Wow, thanks for the write-up! Too bad it didn't solve your OPER button.
I've just added a little bit more clearance around those screw holes, which was quick - will maybe do the 'isolatable button' idea another time 😁

**Comment by gwideman** [comment]
> Wow, thanks for the write-up! Too bad it didn't solve your OPER button.

Yes, but at least we discovered the issue of the non-conducting key pads expeditiously, and that info may help others.

> I've just added a little bit more clearance around those screw holes, which was quick
Great!

> - will maybe do the 'isolatable button' idea another time 😁

It's probably only useful on the likely-to-be-unused buttons, as these are the ones that you might want to repurpose and jumper to others. The patterns associated with defective-pad buttons can stay connected. 

It would be better if someone discovered how to rejuvenate or replace the conductive pads.
